### PR TITLE
Side nav

### PR DESCRIPTION
**Pull request by sethladd**
Needs some testing.

### CODE REVIEW

**Comment by sethladd** [comment]
Fixes https://github.com/flutter/flutter/issues/8283

**Comment by sethladd** [comment]
@InMatrix @LarkAscending @mit-mit ptal

**Comment by sethladd** [comment]
Fixes https://github.com/flutter/flutter/issues/9052

File: _includes/sidebar.html [hunk 1]
```diff
@@ -1,39 +1,53 @@
...
-    {% endfor %}
+<ul id="mysidebar" class="nav">
+
+  <li class="sidebar-title">Get started</li>
+
+    <ul class="sidebar-items">
+      <li><a href="/setup/">Setup and install</a></li>
+      <li><a href="/getting-started/">Create and run your first app</a></li>
+      <li><a href="/bootstrap-into-dart/">Bootstrap into Dart</a></li>
+      <li><a href="/design-principles/">Design principles</a></li>
```
Comment (mit-mit) [inline _includes/sidebar.html]: Why is /design-principles/ under 'Get started'? That page is only relevant for Flutter contributors (ASAICT), which I would assume is a small fraction of general Flutter users/developers.

I suggest we remove it from sidenav, and link from FAQ ("How can I contribute to Flutter").

File: _includes/sidebar.html [hunk 2]
```diff
@@ -1,39 +1,53 @@
...
-
-    {% endif %}
-    {% endfor %}
-    {% endfor %}
+<ul id="mysidebar" class="nav">
+
+  <li class="sidebar-title">Get started</li>
+
+    <ul class="sidebar-items">
+      <li><a href="/setup/">Setup and install</a></li>
```
Comment (mit-mit) [inline _includes/sidebar.html]: 'Install and Setup' to be more chronological?

File: _includes/sidebar.html [hunk 3]
```diff
@@ -1,39 +1,53 @@
...
+      <li><a href="/getting-started/">Create and run your first app</a></li>
+      <li><a href="/bootstrap-into-dart/">Bootstrap into Dart</a></li>
+      <li><a href="/design-principles/">Design principles</a></li>
+      <li><a href="https://docs.google.com/presentation/d/1cw7A4HbvM_Abv320rVgPVGiUP2msVs7tfGbkgdrTy0I/edit?usp=sharing">Architecture diagrams</a></li>
+    </ul>
+
+  <li class="sidebar-title">Build UIs</li>
```
Comment (mit-mit) [inline _includes/sidebar.html]: I think there are too many links in this section; it's a little unwildy

File: _includes/sidebar.html [hunk 4]
```diff
@@ -1,39 +1,53 @@
...
+    <ul class="sidebar-items">
+      <li><a href="/widgets-intro/">Tour the framework</a></li>
+      <li><a href="/widgets/">Widget and component catalog</a></li>
+      <li><a href="/tutorials/layout/">Build Layouts - Tutorial</a></li>
+      <li><a href="/tutorials/interactive/">Add Interactivity - Tutorial</a></li>
+      <li><a href="/web-analogs/">HTML/CSS patterns in Flutter</a></li>
+      <li><a href="/gestures/">Gestures</a></li>
+      <li><a href="/animations/">Animations</a></li>
+      <li><a href="/custom-fonts/">Custom fonts</a></li>
+      <li><a href="/layout/">Box constraints for layout</a></li>
```
Comment (mit-mit) [inline _includes/sidebar.html]: Simplify to `>Box constraints<` as this sidebar section is all about layout

File: _includes/sidebar.html [hunk 5]
```diff
@@ -1,39 +1,53 @@
...
+      <li><a href="/bootstrap-into-dart/">Bootstrap into Dart</a></li>
+      <li><a href="/design-principles/">Design principles</a></li>
+      <li><a href="https://docs.google.com/presentation/d/1cw7A4HbvM_Abv320rVgPVGiUP2msVs7tfGbkgdrTy0I/edit?usp=sharing">Architecture diagrams</a></li>
+    </ul>
+
+  <li class="sidebar-title">Build UIs</li>
+
+    <ul class="sidebar-items">
+      <li><a href="/widgets-intro/">Tour the framework</a></li>
+      <li><a href="/widgets/">Widget and component catalog</a></li>
```
Comment (mit-mit) [inline _includes/sidebar.html]: I'd remove 'and component', that word is fuzzy enough that it confuses more than it clarifies

File: _includes/sidebar.html [hunk 6]
```diff
@@ -1,39 +1,53 @@
...
+
+    <ul class="sidebar-items">
+      <li><a href="/platform-plugins/">Plugins &amp; platform-specific code</a></li>
+      <li><a href="/platform-channels/">Write custom platform-specific code</a></li>
+      <li><a href="/reading-writing-files/">Read and write files</a></li>
+      <li><a href="/networking/">Make network and HTTP requests</a></li>
+    </ul>
+
+
+  <li class="sidebar-title">Test, deploy, workflows</li>
```
Comment (mit-mit) [inline _includes/sidebar.html]: The IntelliJ page (https://flutter.io/intellij-ide/) is missing

File: _includes/sidebar.html [hunk 7]
```diff
@@ -1,39 +1,53 @@
...
+      <li><a href="https://www.youtube.com/watch?v=dkyY9WCGMi0">Framework's layered design <i class="fa fa-video-camera" aria-hidden="true"></i></a></li>
+      <li><a href="https://www.youtube.com/watch?v=UUfXWzp0-DU">Flutter's rendering pipeline <i class="fa fa-video-camera" aria-hidden="true"></i></a></li>
+    </ul>
+
+
+  <li class="sidebar-title">Use device and SDK APIs</li>
+
+    <ul class="sidebar-items">
+      <li><a href="/platform-plugins/">Plugins &amp; platform-specific code</a></li>
+      <li><a href="/platform-channels/">Write custom platform-specific code</a></li>
```
Comment (mit-mit) [inline _includes/sidebar.html]: Simplify to `>Custom platform-specific code<`

File: _includes/sidebar.html [hunk 8]
```diff
@@ -1,39 +1,53 @@
...
+      <li><a href="/widgets-intro/">Tour the framework</a></li>
+      <li><a href="/widgets/">Widget and component catalog</a></li>
+      <li><a href="/tutorials/layout/">Build Layouts - Tutorial</a></li>
+      <li><a href="/tutorials/interactive/">Add Interactivity - Tutorial</a></li>
+      <li><a href="/web-analogs/">HTML/CSS patterns in Flutter</a></li>
+      <li><a href="/gestures/">Gestures</a></li>
+      <li><a href="/animations/">Animations</a></li>
+      <li><a href="/custom-fonts/">Custom fonts</a></li>
+      <li><a href="/layout/">Box constraints for layout</a></li>
+      <li><a href="/assets-and-images/">Use assets and images</a></li>
```
Comment (mit-mit) [inline _includes/sidebar.html]: Simplify to `>Assets and images<`

File: _includes/sidebar.html [hunk 9]
```diff
@@ -1,39 +1,53 @@
...
+    </ul>
+
+  <li class="sidebar-title">Build UIs</li>
+
+    <ul class="sidebar-items">
+      <li><a href="/widgets-intro/">Tour the framework</a></li>
+      <li><a href="/widgets/">Widget and component catalog</a></li>
+      <li><a href="/tutorials/layout/">Build Layouts - Tutorial</a></li>
+      <li><a href="/tutorials/interactive/">Add Interactivity - Tutorial</a></li>
+      <li><a href="/web-analogs/">HTML/CSS patterns in Flutter</a></li>
```
Comment (mit-mit) [inline _includes/sidebar.html]: Remove `in Flutter` part

File: _includes/sidebar.html [hunk 10]
```diff
@@ -1,39 +1,53 @@
...
+    </ul>
+
+
+  <li class="sidebar-title">Use device and SDK APIs</li>
+
+    <ul class="sidebar-items">
+      <li><a href="/platform-plugins/">Plugins &amp; platform-specific code</a></li>
+      <li><a href="/platform-channels/">Write custom platform-specific code</a></li>
+      <li><a href="/reading-writing-files/">Read and write files</a></li>
+      <li><a href="/networking/">Make network and HTTP requests</a></li>
```
Comment (mit-mit) [inline _includes/sidebar.html]: Simplify to `>Networking and HTTP`?

File: _includes/sidebar.html [hunk 11]
```diff
@@ -1,39 +1,53 @@
...
+      <li><a href="/tutorials/layout/">Build Layouts - Tutorial</a></li>
+      <li><a href="/tutorials/interactive/">Add Interactivity - Tutorial</a></li>
+      <li><a href="/web-analogs/">HTML/CSS patterns in Flutter</a></li>
+      <li><a href="/gestures/">Gestures</a></li>
+      <li><a href="/animations/">Animations</a></li>
+      <li><a href="/custom-fonts/">Custom fonts</a></li>
+      <li><a href="/layout/">Box constraints for layout</a></li>
+      <li><a href="/assets-and-images/">Use assets and images</a></li>
+      <li><a href="/text-input/">Handle text input</a></li>
+      <li><a href="https://www.youtube.com/watch?v=dkyY9WCGMi0">Framework's layered design <i class="fa fa-video-camera" aria-hidden="true"></i></a></li>
```
Comment (mit-mit) [inline _includes/sidebar.html]: These two videos seem out of place

File: _includes/sidebar.html (hunk 2 above)
Comment (sethladd) [inline _includes/sidebar.html]: Hah good point.

File: _includes/sidebar.html (hunk 1 above)
Comment (sethladd) [inline _includes/sidebar.html]: Good catch.

File: _includes/sidebar.html (hunk 3 above)
Comment (sethladd) [inline _includes/sidebar.html]: My intention is to land this, and then create some new pages that collect some of these "mid-tier" pages. For example "custom fonts" probably doesn't need to be in the nav.

File: _includes/sidebar.html (hunk 5 above)
Comment (sethladd) [inline _includes/sidebar.html]: The thought was "widget" was fuzzy. But I'll remove component.

File: _includes/sidebar.html (hunk 9 above)
Comment (sethladd) [inline _includes/sidebar.html]: I didn't want to give the impression that you can actually do HTML/CSS in Flutter. But, I'll remove "in Flutter"

File: _includes/sidebar.html (hunk 4 above)
Comment (sethladd) [inline _includes/sidebar.html]: Fair enough

File: _includes/sidebar.html (hunk 8 above)
Comment (sethladd) [inline _includes/sidebar.html]: Done

File: _includes/sidebar.html (hunk 11 above)
Comment (sethladd) [inline _includes/sidebar.html]: I'd like to find a better home in a second sweep through the individual items. For now, I'll leave them in. But agree they might not live here forever.

File: _includes/sidebar.html (hunk 7 above)
Comment (sethladd) [inline _includes/sidebar.html]: Looking at these two titles, I'm still confused what the difference is. Am I supposed to read one, and then the other? What's the difference between Plugins and Custom?

I removed "Write" but I don't understand the relationship between the two docs. :/

File: _includes/sidebar.html (hunk 10 above)
Comment (sethladd) [inline _includes/sidebar.html]: Done

File: _includes/sidebar.html (hunk 6 above)
Comment (sethladd) [inline _includes/sidebar.html]: Good catch

**Comment by sethladd** [comment]
@mit-mit ptal

**Comment by InMatrix** [comment]
LGTM

**Comment by sethladd** [comment]
Thanks. Going to land to get the infrastructure in, and we can continue to iterate on content.

File: _includes/sidebar.html (hunk 7 above)
Comment (mit-mit) [inline _includes/sidebar.html]: Maybe the right solution here is to remove the second page from the sidenav? It's already linked from the first page.

Comment (sethladd) [inline _includes/sidebar.html]: Yeah, that'll work.

**Pull request merged**